src/core/instantiateReactComponent.js

```diff
@@ -40,6 +40,7 @@ assign(
 function isInternalComponentType(type) {
   return (
     typeof type === 'function' &&
+    typeof type.prototype === 'object' &&
```

Comment:
I'm a little curious how this would interact with ES6 classes if we upgrade to ES6.  Is the prototype of an ES6 class a object or a function?  I feel like what you are actually wanting to check is if the prototype is undefined in order to avoid throwing on the next line.